spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -425,14 +431,70 @@ class DLModel[@specialized(Float, Double) T: ClassTag](
   }
 
   override def copy(extra: ParamMap): DLModel[T] = {
-    val copied = new DLModel(model, featureSize, uid).setParent(parent)
+    val copied = new DLModel(_model, featureSize, uid).setParent(parent)
```

Comment:
can we just use model?

Reply:
If  using model, model should be var instead of val, then it will be un-extendable.

Reply:
I don't quite understand why we want to change model in a DLModel.

Reply:
DLClassifierModel.model overiding DLModel.model, if we modify DLModel.model to var, it will lose the ability of be overided (by DLClassifierModel.model).

Reply:
I think modelImpl is better name than _model :)

Reply:
Still I don't understand why model in DLModel needs to be altered.

Reply:
Since we decide to implement [load] by MLReader, there is no need to alter [model] after deserializing [DLModel].